### PR TITLE
`get_pauli_permuted_idx` now returns `size_t`

### PR DESCRIPTION
**Pull request by JonathonMisiewicz**
## Description
The `get_pauli_permuted_idx` utility function now returns a `size_t` rather than an `int` as its first argument. The method's only use is as an argument to `std::next`, which logically should be expecting a `size_t`.

Comments, `auto`, and changing other things to `size_t` to match also included.

## Checklist
- [x] Ready to go!

### CODE REVIEW

File: src/qforte/computer.cc [hunk 1]
```diff
@@ -874,7 +874,7 @@ std::complex<double> Computer::direct_pauli_circ_exp_val(const Circuit& qc) {
     std::vector<int> y_idxs;
     std::vector<int> z_idxs;
 
-    for (const Gate& gate : qc.gates()) {
+    for (const auto& gate : qc.gates()) {
```
Comment (imagoulas) [inline src/qforte/computer.cc:877]: Out of curiosity,  why was this changed to auto?

**Review by imagoulas — APPROVED** [review]
Everything looks good to me. Feel free to merge when you are ready.

File: src/qforte/computer.cc (hunk 1 above)
Comment (JonathonMisiewicz) [reply src/qforte/computer.cc:877]: [Using C++ auto is best practice by default.](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/)
